backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateDAO.java

```diff
@@ -1372,7 +1373,23 @@ public void reverseDefinitiveInvoice(@NonNull final Collection<FlatrateTermId> c
 	@Override
 	public boolean isInvoiceableModularContractExists(@NonNull final IQueryFilter<I_C_Flatrate_Term> filter)
 	{
+		final ICompositeQueryFilter<I_C_Flatrate_Term> purchaseFilter = queryBL
+				.createCompositeQueryFilter(I_C_Flatrate_Term.class)
+				.addEqualsFilter(I_C_Flatrate_Term.COLUMNNAME_IsSOTrx, false)
+				.addEqualsFilter(I_C_Flatrate_Term.COLUMNNAME_IsFinalInvoiced, false);
+
+		final ICompositeQueryFilter<I_C_Flatrate_Term> salesFilter = queryBL
+				.createCompositeQueryFilter(I_C_Flatrate_Term.class)
+				.addInArrayFilter(I_C_Flatrate_Term.COLUMNNAME_IsReadyForDefinitiveInvoice);
```

Comment:
don't we have to check the IsSOTrx flag in this case?

Reply:
yes, thx